clients/venice-client/src/main/java/com/linkedin/venice/fastclient/meta/RequestBasedMetadata.java

```diff
@@ -72,10 +76,16 @@ public class RequestBasedMetadata extends AbstractStoreMetadata {
   private final String clusterDiscoveryD2ServiceName;
   private final ClusterStats clusterStats;
   private final FastClientStats clientStats;
+  private final RouterBackedSchemaReader metadataResponseSchemaReader;
   private volatile boolean isServiceDiscovered;
   private volatile boolean isReady;
+  private Schema metadataResponseSchema = null;
+  private int metadataResponseSchemaId = INVALID_VALUE_SCHEMA_ID;
 
-  public RequestBasedMetadata(ClientConfig clientConfig, D2TransportClient transportClient) {
+  public RequestBasedMetadata(
+      ClientConfig clientConfig,
+      D2TransportClient transportClient,
+      RouterBackedSchemaReader metadataResponseSchemaReader) {
```

Comment:
In case we continue to keep this as a constructor param, we need not restrict it to `RouterBackedSchemaReader`. It can be a generic `SchemaReader` too

Reply:
Will move the creation into `RequestBasedMetadata` and have package private setter method for testing.